qiskit_nature/runtime/vqe_client.py

```diff
@@ -263,7 +268,13 @@ def compute_minimum_eigenvalue(
         # try to convert the operators to a PauliSumOp, if it isn't already one
         operator = _convert_to_paulisumop(operator)
         if aux_operators is not None:
-            aux_operators = [_convert_to_paulisumop(aux_op) for aux_op in aux_operators]
+            try:
+                aux_operators = {
+                    aux_op_name: _convert_to_paulisumop(aux_op)
+                    for aux_op_name, aux_op in aux_operators.items()  # type: ignore
+                }
+            except AttributeError:
+                aux_operators = [_convert_to_paulisumop(aux_op) for aux_op in aux_operators]
```

Comment:
You could also look at the implementation in the `QubitConverter` which handles list or dict scenarios using `qiskit_nature.ListOrDict`. I do not think the code necessarily needs this here, but for consistency purposes we could stick to one way of handling this scenario.
Here is a link to the example section: https://github.com/Qiskit/qiskit-nature/blob/72d5445d59a0829ec5a65f1ba0e36be26d63e63f/qiskit_nature/converters/second_quantization/qubit_converter.py#L325-L336

Reply:
I added a new implementation with `ListOrDict` from utils. It looks cleaner but has the disadvantage that the (deprecated) list type is now converted to a dict by the `ListOrDict` converter without a nice way to restore the initial list after retrieving `job.result()`. So, it's now a dict with strings of numbers 0,1,... as keys (from `ListOrDict`'s `enumerate()`. If that's fine with you, I would like to stick with the cleaner approach using `ListOrDict`.